Dark mode feature

abhiShandy opened the pull request:
Addresses #6 

- used orange shade of Mynode logo (not the bitcoin one)
- added darkmode file just like quicksync
- changed the black parts of logo to orange
- cloned mynode.css to mynode_dark.css with modified font and background colors

Right now it works only for the settings page.

PFA the screenshot of the page
![Screen Shot 2019-10-16 at 10 06 48 PM](https://user-images.githubusercontent.com/6998593/66972609-1d914900-f063-11e9-8f9d-0a77871dd777.png)

abhiShandy commented:
Now, it works for other pages too, but I haven't tested every page. (we need testing framework)

I created a include html for both logo_header and mynode_css, but I had to copy-paste it to every template html :/

I took the long road of using a function in device_info.py and then passing it through templateData for render_template function, but I hope we refactor the code later to use some kind of global variable.
![Screen Shot 2019-10-17 at 12 59 06 AM](https://user-images.githubusercontent.com/6998593/66978839-6eac3780-f079-11e9-8fd7-8266a85a6d06.png)

tehelsper commented:
Thanks for your work so far!

Instead of having one toggle for dark mode, could you change it to get a general UI settings object with dark mode being one setting within the object? That way, we can more easily expand UI settings without needing to continue adding more settings to each of the template parameters.

Other comments:
- Can the button text be a darker color similar to the background?
- Instead of orange text, could it be a bit lighter - like closer to a white or off-white? I'm just wondering how that would look.

abhiShandy commented:
Are you testing this?

tehelsper commented:
Not yet, just looked at screenshots.

abhiShandy commented:
![Screen Shot 2019-10-17 at 10 58 20 PM](https://user-images.githubusercontent.com/6998593/67062573-b2627800-f131-11e9-84ac-e639fd3578ea.png)

check out the colors

abhiShandy commented:
I'll save all the UI settings in one file: `/mnt/hdd/mynode/settings/ui.json`

abhiShandy commented:
You might not like this level of refactoring but
- I'm using ui.json to save UI settings
- I moved the links to CSS and JS files to `includes/head.html`
- changed colors of elements in cli and bitcoind pages, like tables, search fields and cli contents
- tested all pages manually (is there a automated test?)
- couldn't import ui_settings into  electrum_server.py, so that page is stuck in lightmode

tehelsper commented:
The latest screenshot looks great! Most of those changes sound fine, but I'll have to take a look once I get a bit more time.

Any idea what the issue is on the electrum page?

I don't have any automated tests yet, but I'd like to add some. I'm working on a VM that's in beta which I'm hoping will help automate testing since it can easily be spun up from scratch.

tehelsper commented on rootfs/standard/var/www/mynode/templates/includes/head.html:
```diff
@@ -0,0 +1,15 @@
+<link rel="shortcut icon" href="{{ url_for('static', filename='images/favicon.ico') }}">
+<link href="{{ url_for('static', filename='jquery_ui/jquery-ui.css')}}" rel="stylesheet">
+{% if ui_settings['darkmode'] %}
```
To prevent code duplication, would it be possible to make the dark mode a separate CSS file that gets loaded on top of the existing CSS file?

Like

include normal css
if darkmode:
    include dark mode css

Where the dark mode css only has the changes to make the UI dark.

abhiShandy replied:
that will be ideal, I'll try it out and push it in few days

abhiShandy replied:
Done!

tehelsper commented on rootfs/standard/var/www/mynode/templates/lnd.html:
```diff
@@ -1,15 +1,7 @@
-        <script src="{{ url_for('static', filename="js/jquery-3.3.1.min.js")}}"></script>
-        <script src="{{ url_for('static', filename="jquery_ui/jquery-ui.js")}}"></script>
-        <script src="{{ url_for('static', filename="js/mynode.js")}}"></script>
+        {% include 'includes/head.html' %}
```
Nice, I like this.

tehelsper commented on rootfs/standard/var/www/mynode/settings.py:
```diff
@@ -121,6 +120,45 @@ def upgrade_device():
+
+    return ui_settings
+
+def write_ui_settings(ui_settings):
```
You might want to wrap this in a try since the HDD write will fail if the setting is changed before an HDD is detected.

abhiShandy commented:
Changes made:
- merged all changes since 0.1.58
- removed repeated CSS code for dark mode
- change colors of login and config pages for dark mode
- moved contents of bitcoin_cli.css to mynode.css, and deleted bitcoin_cli.css

Unable to import `read_ui_settings` from `settings` into `electrum_server.py`, I can't imagine the reason behind this since the same `import` line works in other `.py` files.
Right now the electrum page is stuck in light mode

abhiShandy commented:
![Screen Shot 2019-11-05 at 9 27 02 PM](https://user-images.githubusercontent.com/6998593/68263087-1bdbf500-0013-11ea-8b02-0d5893b6f50b.png)

Amiga500 commented:
I like it a lot!
